src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/error/AbstractExpectedError.groovy

```diff
@@ -0,0 +1,39 @@
+package org.openkilda.functionaltests.error
+
+import org.openkilda.messaging.error.MessageError
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpClientErrorException
+
+import java.util.regex.Pattern
+
+abstract class AbstractExpectedError extends HttpClientErrorException{
+    HttpStatus statusCode
+    String message
+    Pattern descriptionPattern
+
+    AbstractExpectedError(HttpStatus statusCode, String message, Pattern messagePattern) {
+        super(statusCode)
+        this.statusCode = statusCode
+        this.message = message
+        this.descriptionPattern = messagePattern
+    }
+
+    boolean equals(other) {
```

Comment:
Will work, but comes with some compromises as well:
- you a forced to use `expectedError == actualError` and can't do `actualError == expectedError` (which is not obvious at all)
- also this comes a little bit out of order with all verifications in the code base, where we usually put 'actual' on the lhs of equals expression.

As a quick fix to existing design, I would suggest instead of overriding the 'equals' operator just rename it to sth like `matches`/`conforms`/`satisfies` etc. This will eliminate ambiguity resulting from the uncommon usage of equals operator, since usually it is expected to be 'symmetric'

Reply:
I can't imagine the cases when this standard approach to testing can become ambiguous, and '==' is just shorter than .methodName(). But you're, probably, right. If I refactor to use custom method, there won't be need to inherit from HTTPClientErrorException and would make method signature more strict.

Reply:
> this standard approach to testing

- '==' not being symmetric is not standard 
- 'actual' being on the 'right' is also not standard (at least in this codebase)
- hack with `extends HttpClientErrorException` adds to confusion as well

But I understand what you tried to do, just that it adds more unique approach to asserting rather than making it simpler to a wider auditory. I'm not insisting on any changes though, just suggesting, it's your call. I don't think that this is critical in any way

Reply:
Also, it's a good idea to add javadocs/comments to any hack/tricky solution

Reply:
Overriding 'equals()' is a standard approach. This exact implementation doesn't fully implement it. It lacks spring web client listener that would convert HttpClientErrorException into ExpectError before.